### ISSUE TITLE
Getting "The file does not match your project config" despite having `tsconfig.eslint.json` target the problematic files

### ISSUE THREAD

**Issue by kylemh** [typescript-eslint/typescript-eslint#2297]
Firstly, I want to say I've read the FAQ (on both monorepos and typed linting), plus I spent a lot of time in the following closely-related issues: https://github.com/typescript-eslint/typescript-eslint/issues/1928 and https://github.com/typescript-eslint/typescript-eslint/issues/1723

Also, I wont @ you cuz I'm sure your notifications are wild, but holy shit brad - you're on top of this repo 😂 . You're top 3 active project maintainers I've ever seen for sure.

**Repro**

Sorry, I can't put the repo into a few lines of code because there were some variables, but I think I've made the surface area pretty small! It's a `yarn workspace` monorepo with the ESLint config as a package which is extended as a relative file path from the root config.

https://github.com/kylemh/typescript-eslint-issue

**Expected Result**

- I can lint all TypeScript with type-aware lint rules + my non type-aware lint rules
- I can lint all my JavaScript with non type-aware lint rules

**Actual Result**

- I cannot lint _some_ of my TypeScript

```sh
$ yarn lint
yarn run v1.22.4
$ yarn eslint "**/*.{js,jsx,ts,tsx,mdx}"
$ /Users/kyle/Code/@kylemh/typescript-eslint-issue/node_modules/.bin/eslint '**/*.{js,jsx,ts,tsx,mdx}'

/Users/kyle/Code/@kylemh/typescript-eslint-issue/packages/component-library/src/index.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: packages/component-library/src/index.tsx.
The file must be included in at least one of the projects provided

/Users/kyle/Code/@kylemh/typescript-eslint-issue/packages/component-library/test/blah.test.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: packages/component-library/test/blah.test.tsx.
The file must be included in at least one of the projects provided
```

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `latest` |
| `TypeScript`                | `latest` |
| `ESLint`                    | `latest` |
| `node`                      | `latest` |
| `yarn`                       | `1.18.0` |

**Comment by Retsam** [comment]
Looks like the issue may be how files are specified in your `tsconfig.json`.  You've got:

```
"include": ["**/*.{ts,tsx,mdx}"],
```

But I believe tsconfig isn't looking for that sort of glob syntax.  It works as expected if you replace this with `"include": ["."]` or `"include": ["packages"]`.

**Comment by Retsam** [comment]
Another bit of evidence that it's the `include`, and not anything really to do with `eslint`: with the current settings TS also won't pick up `.d.ts` files as expected.  For example, if you add:

```ts
// packages/component-library/src/whatever.d.ts
declare module 'whatever'; // declare a "whatever" module as `any`
```

Then `import whatever from "whatever"` should compile.  But it doesn't, unless you change the `include` setting.

**Comment by kylemh** [comment]
🤦  thanks so much!
